src/test/java/com/google/sps/GetTop3GemsTest.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the \"License\");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an \"AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import org.junit.Assert;
+import com.google.gson.Gson;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import com.google.sps.servlets.GetTop3GemsServlet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.data.HiddenGem;
+import com.google.sps.data.HiddenGems;
+
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+
+@RunWith(JUnit4.class)
+public final class GetTop3GemsTest {
+
+    @Test
+    public void testServlet() throws Exception {
+      HttpServletRequest request = mock(HttpServletRequest.class);       
+      HttpServletResponse response = mock(HttpServletResponse.class);  
+
+      GetTop3GemsServlet  gemsServlet = new GetTop3GemsServlet();
+
+      StringWriter servletResponse = new StringWriter();
+
+      PrintWriter writer = new PrintWriter(servletResponse);
+      when(response.getWriter()).thenReturn(writer);
+
+      gemsServlet.doGet(request, response);
+      
+      String expected = "[{\"id\":1,\"name\":\"Thomas St. Kitchen \\u0026 Bar\",\"business_type\":\"restaurant\",\"address\":\"169-179 Thomas St, Sydney NSW 2000\",\"price_level\":1,\"rating\":4.2,\"photo\":\"https://cdn.pixabay.com/photo/2015/09/02/12/43/meal-918639_960_720.jpg\"},"
```

Comment:
nit: would probably be nice to format this in a slightly more human readable way, even though it means adding a bunch more + in there.
Something like:
"[{\"id\":1,"
+"\"name\": etc"

That being said if you'd prefer to just have the gson test part below that checks there are 3 responses I'd be fine with that too, we can probably look at adding an equality function to HiddenGem later that would mean you can do the equivalent of the string comparison but with actually objects (don't worry about doing that in this PR though it's a bit involved apparently if you write a .equals function you also have to write a .hashcode or everything breaks)

Reply:
Done. ✅